lib/compat/wordpress-6.3/kses.php

```diff
@@ -7,21 +7,23 @@
  * @package gutenberg
  */
 
-/**
- * Mark CSS safe if it contains grid functions
- *
- * This function should not be backported to core.
- *
- * @param bool   $allow_css Whether the CSS is allowed.
- * @param string $css_test_string The CSS to test.
- */
-function allow_grid_functions_in_styles( $allow_css, $css_test_string ) {
-	if ( preg_match(
-		'/^grid-template-columns:\s*repeat\([0-9,a-z-\s\(\)]*\)$/',
-		$css_test_string
-	) ) {
-		return true;
+if ( ! function_exists( 'allow_grid_functions_in_styles' ) ) {
+	/**
+	 * Mark CSS safe if it contains grid functions
+	 *
+	 * This function should not be backported to core.
+	 *
+	 * @param bool   $allow_css Whether the CSS is allowed.
+	 * @param string $css_test_string The CSS to test.
+	 */
+	function allow_grid_functions_in_styles( $allow_css, $css_test_string ) {
```

Comment:
This one doesn't exist in core, because we patched [kses directly](https://github.com/WordPress/wordpress-develop/pull/4637). I guess it could be prefixed with `gutenberg` instead and then it wouldn't need the `function_exists` check? (there may be other functions in `lib/` to which the same applies)

Reply:
> I guess it could be prefixed with gutenberg instead and then it wouldn't need the function_exists check?

Been wondering about that too. Imho as a "golden rule of thumb" it would probably be safer to always do function_exists()/class_exists() in the plugin's code, no matter what! Scrapping WP betas because of old versions of Gutenberg having conflicting function/class names is no fun :)

Reply:
That's a good point! I don't _think_ renaming would break back compat because this function only exists in the plugin, and there are no matches other than gutenberg in the [wp directory](https://www.wpdirectory.net/search/01H5RNCP1E7ASEY8ZTD81H47JY). But yeah, it doesn't really matter and if leaving it as is can have an educational value, I'm happy with that!